File: DotNet/App/queryHandlers/SearchPostByTextQueryHandler.java
package info.dddeurope.lab.app.queryHandlers;

//import info.dddeurope.lab.app.queries.SearchPostByTextQuery;
//import info.dddeurope.lab.app.views.PostsIndex;
//import lombok.AllArgsConstructor;
//import lombok.Data;
import org.apache.lucene.document.Document;
import java.util.List;

//@Data
//@AllArgsConstructor
public class SearchPostByTextQueryHandler {

    PostsIndex index;

    public List<Document> handle(SearchPostByTextQuery query) throws Exception {
        return this.index.searchTerm(query.getSearchTerm());
    }
}

File: DotNet/App/queryHandlers/SearchPostByTextQueryHandler.cs
using System.Collections.Generic;
using App.Queries;
using App.Views;
using Lucene.Net.Documents;

namespace App.QueryHandlers
{
    public class SearchPostByTextQueryHandler
    {
        readonly PostsIndex _index;

        public SearchPostByTextQueryHandler(PostsIndex index)
        {
            _index = index;
        }

        public List<Document> Handle(SearchPostByTextQuery query)
        {
            return _index.SearchTerm(query.SearchTerm);
        }
    }
}